### PR TITLE
feat(app-platform): Isolate Component Prop Failure

### PR DESCRIPTION
**Pull request by mnoble**
When a single component fails to be prepared (usually because the request to the service fails), don't fail the entire request causing all other components to not render.

### CODE REVIEW

File: src/sentry/api/endpoints/sentry_app_components.py [hunk 1]
```diff
@@ -43,13 +44,15 @@ def get(self, request, organization):
...
-            components.extend(_components)
+                try:
+                    sentry_app_components.Preparer.run(
+                        component=component,
+                        install=install,
+                        project=project,
+                    )
+                    components.append(component)
+                except APIError:
+                    continue
```
Comment (markstory) [inline src/sentry/api/endpoints/sentry_app_components.py]: Would a metric be useful here? Could help in putting partners on the naughty list.

Comment (mnoble) [reply src/sentry/api/endpoints/sentry_app_components.py]: Yea, absolutely. Good call.

Comment (mnoble) [reply src/sentry/api/endpoints/sentry_app_components.py]: Actually, we would have logs for when the downstream object fails. So we could derive this information from that. I don't know if we need to explicitly do something here.

Comment (markstory) [reply src/sentry/api/endpoints/sentry_app_components.py]: Sounds good. :+1:

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**